src/librustc/ty/layout.rs

```diff
@@ -356,33 +372,37 @@ impl Integer {
             (I16, false) => tcx.types.u16,
             (I32, false) => tcx.types.u32,
             (I64, false) => tcx.types.u64,
+            (I128, false) => tcx.types.u128,
             (I1, true) => tcx.types.i8,
             (I8, true) => tcx.types.i8,
             (I16, true) => tcx.types.i16,
             (I32, true) => tcx.types.i32,
             (I64, true) => tcx.types.i64,
+            (I128, true) => tcx.types.i128,
         }
     }
 
     /// Find the smallest Integer type which can represent the signed value.
     pub fn fit_signed(x: i64) -> Integer {
```

Comment:
@est31 handling 128 bit discriminants is not exactly trivial to implement and might even involve tweaking LLVM, as it does not have a way to provide debug info for 128 bit discriminants.. That’s why it was left off for an initial implementation.